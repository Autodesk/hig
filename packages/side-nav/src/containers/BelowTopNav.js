import React, { Component } from "react";
import PropTypes from "prop-types";
import Surface from "@hig/surface";
import ThemeContext from "@hig/theme-context";
import stylesheet from "./stylesheet";

export default class BelowTopNav extends Component {
  static propTypes = {
    /** A SideNav element */
    children: PropTypes.node,
    /** Called when beginning to hover over the container */
    onMouseEnter: PropTypes.func,
    /** Called when no longer hovering over the container */
    onMouseLeave: PropTypes.func,
    /** Function to modify the component's styles */
    stylesheet: PropTypes.func
  };

  render() {
    const {
      children,
      onMouseLeave,
      onMouseEnter,
      stylesheet: customStylesheet
    } = this.props;

    return (
      <ThemeContext.Consumer>
        {({ resolvedRoles }) => (
          <Surface
            onMouseEnter={onMouseEnter}
            onMouseLeave={onMouseLeave}
            style={
              stylesheet(
                { stylesheet: customStylesheet, ...this.props },
                resolvedRoles
              ).belowTop
            }
          >
            {children}
          </Surface>
        )}
      </ThemeContext.Consumer>
    );
  }
}
