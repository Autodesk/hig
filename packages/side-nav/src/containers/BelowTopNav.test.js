import React from "react";
import renderer from "react-test-renderer";

import BelowTopNav from "./BelowTopNav";

describe("side-nav/containers/BelowTopNav", () => {
  describe("snapshot tests", () => {
    const cases = [
      {
        description: "renders without props",
        props: {},
      },
    ];

    cases.forEach(({ description, props: { children, ...otherProps } }) => {
      it(description, () => {
        const wrapper = <BelowTopNav {...otherProps}>{children}</BelowTopNav>;
        const tree = renderer.create(wrapper).toJSON();

        expect(tree).toMatchSnapshot();
      });
    });
  });
});
