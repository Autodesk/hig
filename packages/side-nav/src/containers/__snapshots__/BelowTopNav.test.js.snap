// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`side-nav/containers/BelowTopNav snapshot tests renders without props 1`] = `
<div
  className="hig--light-theme hig__side-nav-container--below-top-nav"
/>
`;
